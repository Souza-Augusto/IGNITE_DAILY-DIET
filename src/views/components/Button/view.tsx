import { ReactNode } from 'react';

import { TouchableOpacityProps } from 'react-native';
import { ButtonTypeStyleProps, Container, Title } from './styles';

type Props = TouchableOpacityProps & {
  title: string;
  before?: ReactNode;
  type?: ButtonTypeStyleProps;
};

export function Button({
  before,
  type = 'PRIMARY',
  title,
  style,
  ...rest
}: Props) {
  return (
    <Container type={type} style={style} activeOpacity={0.5} {...rest}>
      {before}
      <Title type={type}>{title}</Title>
    </Container>
  );
}
