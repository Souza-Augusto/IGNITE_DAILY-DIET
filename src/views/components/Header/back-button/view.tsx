import { BackIconContainer, BackIcon } from './styles';
import { useBackButtonViewModel } from './view-model';

interface Props {
  color?: string;
}

export function BackButton({ color }: Props) {
  const { handleGoBack } = useBackButtonViewModel();

  return (
    <BackIconContainer testID='back-button' onPress={handleGoBack}>
      <BackIcon name='arrowleft' color={color} />
    </BackIconContainer>
  );
}
