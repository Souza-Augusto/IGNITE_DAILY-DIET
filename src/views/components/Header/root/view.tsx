import { ReactNode } from 'react';
import { Container } from './styles';
import { useNavigation } from '@react-navigation/native';

type Props = {
  leftElement?: ReactNode;
  rightElement?: ReactNode;
  children?: ReactNode;
  backButton?: boolean;
};

export function Root({ leftElement, rightElement, children }: Props) {
  return (
    <Container>
      {leftElement}
      {children}
      {rightElement}
    </Container>
  );
}
