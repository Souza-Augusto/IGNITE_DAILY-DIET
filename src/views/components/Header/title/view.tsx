import { Container, HeaderTitle } from './styles';

interface Props {
  title: string;
}

export function Title({ title }: Props) {
  return (
    <Container>
      <HeaderTitle>{title}</HeaderTitle>
    </Container>
  );
}
