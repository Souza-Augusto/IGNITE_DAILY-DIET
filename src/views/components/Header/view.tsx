import { Container, Logo, Profile } from './styles';
import { useNavigation } from '@react-navigation/native';

type Props = {
  showBackButton?: boolean;
};

export function Header({ showBackButton = false }: Props) {
  const navigation = useNavigation();

  return (
    <Container>
      <Logo source={require('@assets/images/png/Logo.png')} />
      <Profile source={require('@assets/images/png/Ellipse.png')} />
    </Container>
  );
}
