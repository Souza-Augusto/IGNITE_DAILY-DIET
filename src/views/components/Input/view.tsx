import { Container } from './styles';
import { TextInputProps } from 'react-native';
import theme from '@theme/index';

export function Input({ ...rest }: TextInputProps) {
  return (
    <Container
      selectionColor={theme.COLORS.GRAY_500}
      placeholderTextColor={theme.COLORS.GRAY_400}
      {...rest}
    />
  );
}
