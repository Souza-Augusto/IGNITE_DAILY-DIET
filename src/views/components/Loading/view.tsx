import { Container, LoadIndicator } from './styles';

export function Loading() {
  return (
    <Container>
      <LoadIndicator testID='component-loading' />
    </Container>
  );
}
