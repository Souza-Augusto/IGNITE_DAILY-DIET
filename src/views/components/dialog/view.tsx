import { Modal, ModalProps } from 'react-native';
import {
  ConfirmButton,
  DialogContainer,
  TitleAndButtonContainer,
  DialogTitle,
  DialogButtonsContainer,
  CancelButton,
  CancelButtonTitle,
  ConfirmButtonTitle,
  TranparentButton,
} from './styles';

export interface Props extends ModalProps {
  dialogVisible: boolean;
  dialogMessage: string;
  positiveButtonTitle?: string;
  negativeButtonTitle?: string;
  negativeFunction?: () => void;
  positiveFunction?: () => void;
}

export function Dialog({
  dialogVisible,
  dialogMessage,
  negativeButtonTitle,
  positiveButtonTitle,
  negativeFunction,
  positiveFunction,
  ...rest
}: Props) {
  return (
    <Modal visible={dialogVisible} transparent {...rest}>
      <DialogContainer>
        <TranparentButton onPress={negativeFunction} />
        <TitleAndButtonContainer>
          <DialogTitle>{dialogMessage}</DialogTitle>
          <DialogButtonsContainer>
            {negativeButtonTitle && (
              <CancelButton onPress={negativeFunction}>
                <CancelButtonTitle numberOfLines={1}>
                  {negativeButtonTitle}
                </CancelButtonTitle>
              </CancelButton>
            )}
            {positiveButtonTitle && (
              <ConfirmButton onPress={positiveFunction}>
                <ConfirmButtonTitle>{positiveButtonTitle}</ConfirmButtonTitle>
              </ConfirmButton>
            )}
          </DialogButtonsContainer>
        </TitleAndButtonContainer>
        <TranparentButton onPress={negativeFunction} />
      </DialogContainer>
    </Modal>
  );
}
