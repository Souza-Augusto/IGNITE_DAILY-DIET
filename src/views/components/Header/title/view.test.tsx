import { render } from '@__tests__/utils/customRenderItem';
import { Title } from './view';

describe('component header title', () => {
  it('shoul render correctly', () => {
    const { toJSON, getByText } = render(<Title title='teste...' />);

    const TITLE = getByText('teste...');

    expect(TITLE).toBeDefined();
    expect(toJSON).toMatchSnapshot();
  });
});
