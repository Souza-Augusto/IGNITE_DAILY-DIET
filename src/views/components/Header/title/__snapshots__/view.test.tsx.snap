// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component header title shoul render correctly 1`] = `[Function]`;
